### ISSUE TITLE
.AppImage artifacts: keytar’s native module causes crash (invalid ELF header)

### ISSUE THREAD

**Issue by strogonoff** [glossarist/glossarist-desktop#117]
Full trace:

```
./Glossarist-1.6.22.AppImage

A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot open /tmp/.mount_GlossaZsXNR3/resources/app.asar/keytar.node: Error: /tmp/.mount_GlossaZsXNR3/resources/app.asar.unpacked/keytar.node: invalid ELF header
    at Object.<anonymous> (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/node_modules/keytar/build/Release/keytar.node:1:160)
    at Object.call (/tmp/.mount_GlossaZsXNR3/resources/app.asar/main.js:2:351365)
    at _webpack_require_ (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/webpack/bootstrap:19:22)
    at Object.call (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/node_modules/keytar/lib/keytar.js:1:14)
    at _webpack_require_ (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/webpack/bootstrap:19:22)
    at Module.call (/tmp/.mount_GlossaZsXNR3/resources/app.asar/main.js:2:416592)
    at _webpack_require_ (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/webpack/bootstrap:19:22)
    at Object.call (/tmp/.mount_GlossaZsXNR3/resources/app.asar/main.js:2:141135)
    at _webpack_require_ (/tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/webpack/bootstrap:19:22)
    at /tmp/.mount_GlossaZsXNR3/resources/app.asar/webpack:/webpack/bootstrap:83:10
```

**Comment by strogonoff** [comment]
This on Ubuntu 18.04

**Comment by strogonoff** [comment]
For background, node-keytar requires using a native Node module (bundled as a binary image). We use node-keytar to store Git passwords using user’s native OS mechanisms.

**Comment by strogonoff** [comment]
This may be caused by the fact that the .AppImage was built on macOS, but app ran on Ubuntu. Need to see how to make sure images we build on GHA will work on whatever Linux distributions we support.

**Comment by ronaldtse** [comment]
@webdev778 reports this issue after compiling on Ubuntu (which succeeds), but crashes when run:

![image](https://user-images.githubusercontent.com/11865/92201603-9386e480-eeaf-11ea-8766-e40026d615f6.png)

**Comment by strogonoff** [comment]
Fixed in Snap package in v1.6.26
